src/main/java/core/basesyntax/strategy/StorageReportGenerateImpl.java

```diff
@@ -0,0 +1,17 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.db.Storage;
+import java.util.Map;
+
+public class StorageReportGenerateImpl implements StorageReportGenerate {
+    public String getReport() {
+        StringBuilder report = new StringBuilder();
+        report.append("fruit,quantity").append(System.lineSeparator());
+        for (Map.Entry<String, Integer> entry : Storage.getAllFruits().entrySet()) {
```

Comment:
Direct access to the `Storage` class's methods is not recommended. It would be better to inject a dependency on a service that interacts with `Storage` rather than calling it directly, to adhere to the SOLID principles, particularly the Dependency Inversion Principle.